ui/desktop/src/components/ToolCallWithResponse.tsx

```diff
@@ -26,11 +26,15 @@ interface ToolCallWithResponseProps {
   append?: (value: string) => void; // Function to append messages to the chat
 }
 
-function getToolResultValue(toolResult: Record<string, unknown>): Content[] | null {
-  if ('value' in toolResult && Array.isArray(toolResult.value)) {
-    return toolResult.value as Content[];
+function getToolResultContent(toolResult: Record<string, unknown>): Content[] {
+  if (toolResult.status !== 'success') {
+    return [];
   }
-  return null;
+  const value = toolResult.value as { content: Content[] };
+  return value.content.filter((item) => {
+    const annotations = (item as { annotations?: { audience?: string[] } }).annotations;
+    return !annotations?.audience || annotations.audience.includes('user');
+  });
 }
```

Comment:
this is fine, but casting the value to CallToolResponse from the api would be even better and avoid this sort of problem in the future